src/Authentication/Authentication/Cmdlets/SetMgGraphOption.cs

```diff
@@ -0,0 +1,58 @@
+// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+using Newtonsoft.Json;
+using System.IO;
+using System.Management.Automation;
+
+namespace Microsoft.Graph.PowerShell.Authentication.Cmdlets
+{
+    [Cmdlet(VerbsCommon.Set, "MgGraphOption", HelpUri = "")]
+    public class SetMgGraphOption : PSCmdlet
+    {
+        [Parameter]
+        public SwitchParameter EnableLoginByWAM { get; set; }
+
+        protected override void BeginProcessing()
+        {
+            base.BeginProcessing();
+        }
+
+        protected override void ProcessRecord()
+        {
+            base.ProcessRecord();
+
+            bool currentSetting = GraphSession.Instance.GraphOption.EnableWAMForMSGraph;
+
+            if (EnableLoginByWAM.IsPresent)
+            {
+                GraphSession.Instance.GraphOption.EnableWAMForMSGraph = EnableLoginByWAM;
+                if (GraphSession.Instance.GraphOption.EnableWAMForMSGraph)
+                    WriteObject("[Preview] Signin by Web Account Manager (WAM) is enabled.");
+                else
+                    WriteObject("[Preview] Signin by Web Account Manager (WAM) is disabled.");
+            }
+            else
+            {
+                GraphSession.Instance.GraphOption.EnableWAMForMSGraph = !currentSetting;
+                if (GraphSession.Instance.GraphOption.EnableWAMForMSGraph)
+                    WriteObject("[Preview] Signin by Web Account Manager (WAM) is enabled.");
+                else
+                    WriteObject("[Preview] Signin by Web Account Manager (WAM) is disabled.");
+            }
```

Comment:
No need to toggle WAM when `-EnableLoginByWam` is not set. Using `SwitchParameter` assumes `-EnableLoginByWam` or `-EnableLoginByWam:$true` for true, and `-EnableLoginByWam:$false` or no parameter for false. Is this the behavior you were going for?

```suggestion
```

IMO, calling `Set-MgGraphOption` with no parameter (-EnableLoginByWam is optional) shouldn't change anything in the option object since no property was provided. Would it be better to use a Boolean for `EnableLoginByWam` parameter instead? See https://learn.microsoft.com/en-us/powershell/scripting/developer/cmdlet/types-of-cmdlet-parameters?view=powershell-7.3#switch-parameters:
> Windows PowerShell provides a [System.Management.Automation.SwitchParameter](https://learn.microsoft.com/en-us/dotnet/api/System.Management.Automation.SwitchParameter) type that allows you to define a parameter whose value is automatically set to false if the parameter is not specified when the cmdlet is called. Whenever possible, use switch parameters in place of Boolean parameters.